README.md

```diff
@@ -1,12 +1,54 @@
-# zksync-state-reconstruct
-> Tool / Library to reconstruct zkSync state from commit blocks
+# zkSync State Reconstruction Tool
+> Tool / Library to reconstruct zkSync state from commit blocks published on L1
 
-## Prerequisites
-This tool is written in nightly Rust; you can install Rust by following the official instructions [here](https://www.rust-lang.org/learn/get-started), and then running the following command to switch to the nightly toolchain:
+## Prerequisites & setup
+Currently there are three ways to setup the environment: using the provided Nix flake, the container image, or installing the dependencies manually.
 
-```fish
-rustup toolchain install nightly
+<details>
+  <summary>Nix Flake</summary>
+  To use the supplied Nix development environment you need to have Nix installed, This can be done by following the official instructions [here](https://nixos.org/download.html).
+
+  Once Nix is installed, the development environment can be activated via the following command:
+
+  ```nix
+  nix develop
 ```
+</details>
+
+<details>
+  <summary>Container Image</summary>
+  To build the container image, use:
+
+  ```fish
+  podman build -t state-reconstruction:latest .
+  ```
+
+  And, to run it with `podman`, please use:
+
+  ```fish
+  podman run -it state-reconstruction:latest
+  ```
+</details>
+
+<details>
+  <summary>Manually</summary>
+  This tool is written in nightly Rust; you can install Rust by following the official instructions [here](https://www.rust-lang.org/learn/get-started), and then running the following command to switch to the nightly toolchain:
+
+  ```fish
+  rustup toolchain install nightly
+  ```
+
+  You also need to have a recent version of `protobuf` installed and accessible via `PATH`. Use your preferred package manager to do this. For example, using brew:
+
+  ```fish
+  brew install protobuf
+  ```
+</details>
+
+> Note: it is recommend to override the maximum number of open file descriptors. Without doing so you may eventually run into an error, halting progress. On Unix machines this can be done by using `ulimit` along with the `-n` argument:
+> ```fish
+> ulimit -n 8192
```

Comment:
is that enough? We should probably test the full run, one of these days (which would also provide a space estimate)...

Reply:
I can't say for certain, this is what I'm running on right now after experiencing the error with `ulimit -n` on `1024`. Should that happen again with the current value I'll adjust it as the sync progresses.